### PR TITLE
Application record models

### PR DESCRIPTION
**Pull request by see-why**
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.
Models include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
 -  Create a method called current_user in ApplicationController which returns the first user from a database.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Comment < ApplicationRecord
+  belongs_to :posts
```
Comment (bimbolabuari) [inline app/models/comment.rb]: According to the ERD Diagram, comment `belongs_to` posts and users

Let's add users to the association 👍

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Like < ApplicationRecord
+  belongs_to :posts
```
Comment (bimbolabuari) [inline app/models/like.rb]: According to the ERD Diagram, like `belongs_to` both posts and users

Let's add users to the association 👍

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @see-why, 

You have done a good job so far 🚀  

## Highlights 🎉 

- Create a method called current_user in your ApplicationController.
- Current_user method return the first user from a database.
- Users
   - A method that returns the 3 most recent posts for a given user.
- Posts
  -  A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.
- Comments
   - A method that updates the comments counter for a post.
- Likes
   - A method that updates the likes counter for a post.
  
There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

File: app/models/comment.rb (hunk 1 above)
Comment (see-why) [reply app/models/comment.rb]: thanks for pointing this out.

👍

File: app/models/like.rb (hunk 2 above)
Comment (see-why) [reply app/models/like.rb]: thanks for pointing this out.

👍

**Review by addod19 — APPROVED** [review]
### Hi @see-why, 

Your project is complete! Nice job adding the belongs_to relationship :fireworks:  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**